constantine/eth_verkle_ipa/transcript_gen.nim

```diff
@@ -27,37 +29,72 @@ func newTranscriptGen*(res: var CryptoHash, label: openArray[byte]) =
   res.update(label)
 
 func messageAppend*(res: var CryptoHash, message: openArray[byte], label: openArray[byte]) =
-  res.init()
   res.update(label)
   res.update(message)
 
 func messageAppend_u64*(res: var CryptoHash, label: openArray[byte], num_value: uint64) = 
-  res.init()
   res.update(label)
   res.update(num_value.toBytes(bigEndian))
 
 func domainSeparator*(res: var CryptoHash, label: openArray[byte]) =
-  var state {.noInit.} : CryptoHash
-  state.update(label)
+  res.update(label)
 
 func pointAppend*(res: var CryptoHash, label: openArray[byte], point: EC_P) =
   var bytes {.noInit.}: array[32, byte]
-  if bytes.serialize(point) == cttCodecEcc_Success:
-    res.messageAppend(bytes, label)
+  let stat = bytes.serialize(point) 
+  doAssert stat == cttCodecEcc_Success, "Serialization Failure!"
+  res.messageAppend(bytes, label)
 
 func scalarAppend*(res: var CryptoHash, label: openArray[byte], scalar: matchingOrderBigInt(Banderwagon)) =
   var bytes {.noInit.}: array[32, byte]
 
-  if bytes.serialize_scalar(scalar) == cttCodecScalar_Success:
-    res.messageAppend(bytes, label)
+  let stat = bytes.serialize_scalar(scalar, littleEndian)
+  doAssert stat == cttCodecScalar_Success, "Issues with marshalling!"
+  res.messageAppend(bytes, label)
+
+func fromVerkleDigest(dst: var Fr[Banderwagon], src: array[32, byte]) : bool =
+  ## The input src byte array that we get here is 32 bytes
```

Comment:
"## The input src byte array is a 256-bit integer in little-endian representation"

Reply:
Actually we don't even need to mention integer, we can just say in the doc:
"Convert a 32-bytes digest to a Verkle Trie protocol field element."